webapp-django/crashstats/crashstats/tests/test_models.py

```diff
@@ -335,6 +336,14 @@ def mocked_get(**params):
         assert r['Vendor'] == 'Mozilla'
         assert 'Email' in r  # no filtering at this level
 
+    def test_raw_crash_invalid_id(self):
+        model = models.RawCrash
+        model.implementation = self._mockeries[models.RawCrash]
```

Comment:
Since this line is weird and anyone reading through this code is likely to stop abruptly with a "wait, what?", it's worth adding a comment. Maybe something along the lines of:

```
# NOTE(alexis): this undoes the mocking of the implementation so we can test
# the implementation code.
```

Also, `implementation` is a class property, but what you're doing here is setting an instance property which shadows the class property. I think that's fine in this case, but it's worth knowing that the class property is still a `MagicMock` and if there's code that refers to the property via the class, it won't work. If you create another instance and expect it to have the same value as this one, it won't.

```
$ python
Python 2.7.15rc1 (default, Apr 15 2018, 21:51:34) 
[GCC 7.3.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> class Foo(object):
...     someprop = True
... 
>>> foo = Foo()
>>> foo.someprop = False
>>> bar = Foo()
>>> foo.someprop
False
>>> bar.someprop
True
>>> Foo.someprop
True
>>> 
```

Another way to do this that doesn't involve instance properties could be:

```
# NOTE(alexis): this undoes the mocking of the implementation so we can test
# the implementation code.
models.RawCrash.implementation = self._mockeries[models.RawCrash]
model = models.RawCrash
api = model()
```